containers/invoices/InvoiceState.js

```diff
@@ -4,13 +4,6 @@ import { c } from 'ttag';
 import PropTypes from 'prop-types';
 import { INVOICE_STATE } from 'proton-shared/lib/constants';
 
-const STATES = {
-    [INVOICE_STATE.UNPAID]: c('Invoice state display as badge').t`Unpaid`,
-    [INVOICE_STATE.PAID]: c('Invoice state display as badge').t`Paid`,
-    [INVOICE_STATE.VOID]: c('Invoice state display as badge').t`Void`,
-    [INVOICE_STATE.BILLED]: c('Invoice state display as badge').t`Billed`
-};
```

Comment:
I think I prefer this pattern:

```js
const getStatesI18N = () => ({
    [INVOICE_STATE.UNPAID]: c('Invoice state display as badge').t`Unpaid`,
    [INVOICE_STATE.PAID]: c('Invoice state display as badge').t`Paid`,
    [INVOICE_STATE.VOID]: c('Invoice state display as badge').t`Void`,
    [INVOICE_STATE.BILLED]: c('Invoice state display as badge').t`Billed`
});

const InvoiceState = ({ invoice }) => {
   const i18n = getStatesI18N();
   return <Badge type={i18n[invoice.State]}>{STATES[invoice.State]}</Badge>;
};
```

Reply:
A similar solution (defining the translation strings outside but generating them via a function inside the component) could be used in some other files I retouched. Do you want to use it in other places too (there are a few, but see `LogsTable` or `LogsSection` for examples)?

Reply:
Yes I prefer it generally

Reply:
Done! It looks more homogeneous now